src/jlgen.jl

```diff
@@ -350,7 +350,7 @@ function compile_method_instance(@nospecialize(job::CompilerJob),
     end
 
     # process all compiled method instances
-    compiled = Dict()
+    compiled = Dict{MethodInstance, NamedTuple{(:ci, :func, :specfunc), Tuple{CodeInstance, LLVM.Function, LLVM.Function}}}()
```

Comment:
Typing this seems counter-productive: this code isn't performance sensitive, the values are stored by pointer anyway, and now you have it specialize for this weird combination of key/value that we aren't able to precompile (since it's methods from Base). So just leave this untyped, aka. Any/Any.